http://gitweb.freedesktop.org/?p=xorg/xserver.git;a=commit;h=b5fde366e2e21234ac0b81222fd5c42ca3e49cba
Properly free device devPrivates - memory leak fix.
	author  Eamon Walsh <ewalsh@tycho.nsa.gov>
	Wed, 7 Mar 2007 17:29:55 +0000 (12:29 -0500)
	committer  Eamon Walsh <ewalsh@moss-uranus.epoch.ncsc.mil>
	Wed, 7 Mar 2007 17:29:55 +0000 (12:29 -0500)
	commit  b5fde366e2e21234ac0b81222fd5c42ca3e49cba
	tree  21c4d64fc97f8a7b0ec39f7e7a408540dd78691e  tree
	parent  a3d2c5d622d9ca36d6fa2966aff09524e3ea39ac  commit | commitdiff
Properly free device devPrivates - memory leak fix.

--- xorg-server-1.2.0.orig/dix/devices.c	2007-03-18 21:43:21.000000000 +0100
+++ xorg-server-1.2.0/dix/devices.c	2007-03-18 21:47:58.000000000 +0100
@@ -268,20 +268,24 @@
 	if (l->xkb_sli)
 	    XkbFreeSrvLedInfo(l->xkb_sli);
 #endif
 	xfree(l);
     }
 #ifdef XKB
     while (dev->xkb_interest) {
 	XkbRemoveResourceClient((DevicePtr)dev,dev->xkb_interest->resource);
     }
 #endif
+
+    if (dev->devPrivates)
+        xfree(dev->devPrivates);
+
     xfree(dev->sync.event);
     xfree(dev);
 }
 
 void
 CloseDownDevices()
 {
     register DeviceIntPtr dev, next;
 
     for (dev = inputInfo.devices; dev; dev = next)
